src/vcf/fixer.cpp

```diff
@@ -210,10 +238,149 @@ namespace ebi
+                                               const std::string &separator)
+        {
+            std::set<std::string> already_present;
+            auto is_value_duplicated = [&](const std::string &value, size_t index) -> bool {
+                bool first_appearance = already_present.insert(value).second;
+                return not first_appearance;
+            };
+            return remove_fields(column, separator, is_value_duplicated);
+        }
+
+        size_t Fixer::remove_duplicate_key_value_pairs(const std::string &column,
+                                                       const std::string &separator,
+                                                       const std::string &key_value_separator,
+                                                       const std::string &empty_value)
+        {
+            std::map<std::string, std::string> first_key_occurrence;
+            std::vector<std::string> ordered_keys;
+            std::set<std::string> fields_to_remove;
+
+            std::vector<std::string> fields;
+            util::string_split(column, separator.c_str(), fields);
+
+            for (auto & field : fields) {
+                std::vector<std::string> subfields;
+                util::string_split(field, key_value_separator.c_str(), subfields);
+                auto first_key_iterator = first_key_occurrence.find(subfields[0]);
+                if (first_key_iterator == first_key_occurrence.end()) {
+                    ordered_keys.push_back(subfields[0]);
+                    first_key_occurrence[subfields[0]] = subfields[1];
+                } else if (first_key_occurrence[subfields[0]] != subfields[1]) {
+                    fields_to_remove.insert(subfields[0]);
+                }
+            }
+
+            std::vector<std::string> fixed_column;
+            size_t num_removed_duplicates = 0;
+
+            for (auto & ordered_key : ordered_keys) {
+                if (fields_to_remove.find(ordered_key) == fields_to_remove.end()) {
+                    fixed_column.push_back(ordered_key + key_value_separator + first_key_occurrence[ordered_key]);
+                } else {
+                    num_removed_duplicates++;
+                }
+            }
+
+            if (fixed_column.size() == 0) {       // all the fields were removed
+                output << empty_value;
+            } else {
+                util::print_container(output, fixed_column, "", separator, "");
+            }
+
+            return num_removed_duplicates;
+        }
+
+        size_t Fixer::remove_duplicate_format_sample_pairs(const std::string &string_line) {
+            const size_t format_column_index = 8;
+            size_t num_removed_duplicates = 0;
+
+            using iter = std::vector<std::string>::iterator;
+            fix_columns(format_column_index, -1, string_line, "\t", [&](iter first, iter last) {
+                std::map<std::string, std::vector<size_t>> format_fields_indexes;
```

Comment:
Yes, we need to use the other positions in the method `get_format_fields_to_remove` (https://github.com/Anishka0107/vcf-validator/blob/debug_new/src/vcf/fixer.cpp#L367)😅
The vector here indicates the indices where the duplicated field is present in the format column. So we have to look at the same indices in the sample columns to check if each column has the same or different values for that field.